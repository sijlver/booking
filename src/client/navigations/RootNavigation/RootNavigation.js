import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';

import { HomePage, AboutPage, NotFoundPage, HotelPage } from '../../pages';

class RootNavigation extends Component {
    render() {
        return(
            <div>
                <Switch>
                    <Route exact path='/' component={HomePage} />
                    <Route path='/about' component={AboutPage} />
                    <Route path='/hotel/:id' component={HotelPage} />
                    <Route path='*' component={NotFoundPage} />
                </Switch>
            </div>
        );
    }
}

export default RootNavigation;
