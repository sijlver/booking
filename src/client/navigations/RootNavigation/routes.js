import { HomePage, AboutPage, HotelPage, NotFoundPage } from '../../pages';

const routes = [
    {
        path: '/',
        exact: true,
        component: HomePage,
    },
    {
        path: '/about',
        component: AboutPage,
    },
    {
        path: '/hotel/:id',
        component: HotelPage,
    },
    {
        path: '*',
        component: NotFoundPage,
    },
];

export default routes;
