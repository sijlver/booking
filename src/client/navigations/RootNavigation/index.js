import RootNavigation from './RootNavigation';
import routes from './routes';

export { RootNavigation, routes };
