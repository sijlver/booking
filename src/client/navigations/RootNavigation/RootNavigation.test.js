import React from 'react';
import { shallow  } from "enzyme";
import jest from 'jest-mock';

import RootNavigation from './RootNavigation';

describe('RootNavigation', () => {
    it('should render self', () => {
        const wrapper = shallow (
            <RootNavigation />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
