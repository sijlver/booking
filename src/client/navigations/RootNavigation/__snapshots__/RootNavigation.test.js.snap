// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootNavigation should render self 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RootNavigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/about"
        />
        <Route
          component={[Function]}
          path="/hotel/:id"
        />
        <Route
          component={[Function]}
          path="*"
        />
      </Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            path="/about"
          />,
          <Route
            component={[Function]}
            path="/hotel/:id"
          />,
          <Route
            component={[Function]}
            path="*"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/about",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/hotel/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "*",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/about"
          />
          <Route
            component={[Function]}
            path="/hotel/:id"
          />
          <Route
            component={[Function]}
            path="*"
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              path="/about"
            />,
            <Route
              component={[Function]}
              path="/hotel/:id"
            />,
            <Route
              component={[Function]}
              path="*"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/about",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/hotel/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "*",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
